### PR TITLE
MDEXP-151 Mapping profile - transformations - process holdings perman…

### PR DESCRIPTION
**Pull request by illia-borysenko**
## Description
Update RuleFactory to start creating rules by transformations with translation functions. Translation functions are assigned to specific transformation fieldId and stored in static Map.

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

**Comment by KVupp** [comment]
@illia-borysenko  It is fine for now, as we are in a time crunch, but Please do not paste the description from the ticket , we can read it from there :) This PR description could be precise just to know what changes went in. It will be helpful in the future if we are debugging something to know why it was done in a particular way. Thanks!

File: src/main/java/org/folio/clients/InventoryClient.java [hunk 1]
```diff
@@ -42,6 +43,11 @@
     return getReferenceDataByUrl(endpoint, params, IDENTIFIER_TYPES);
   }
 
+  public Map<String, JsonObject> getLocations(OkapiConnectionParams params) {
```
Comment (KVupp) [inline src/main/java/org/folio/clients/InventoryClient.java:46]: Permissions needs to be added in module descriptor

File: src/main/java/org/folio/clients/InventoryClient.java [hunk 2]
```diff
@@ -42,6 +43,11 @@
...
+    String endpoint = resourcesPathWithPrefix(LOCATIONS) + "?limit=" + REFERENCE_DATA_LIMIT;
+    return getReferenceDataByUrl(endpoint, params, LOCATIONS);
+  }
```
Comment (KVupp) [inline src/main/java/org/folio/clients/InventoryClient.java:49]: Also please update Integration tests in DataExportTest

File: src/main/java/org/folio/clients/InventoryClient.java (hunk 1 above)
Comment (illia-borysenko) [reply src/main/java/org/folio/clients/InventoryClient.java:46]: Updated

File: src/main/java/org/folio/clients/InventoryClient.java (hunk 2 above)
Comment (illia-borysenko) [reply src/main/java/org/folio/clients/InventoryClient.java:49]: Updated

**Pull request merged**